### PR TITLE
More phpstan/psalm fixes

### PR DESCRIPTION
**Pull request by dereuromark**
Most importantly, the helper wasnt in sync with the lib.
But also the |bool are a lie as only dirty false is also returned.
If you otherwise check for false return via `if ()` the tools will always complain that true wasnt handled, but it shouldnt be as true is not a valid return value.

### CODE REVIEW

**Comment by dereuromark** [comment]
I cant resolve the

    Call to an undefined method Cake\ORM\Association::junction().

error, even though I use inline annotations
Do we need to exclude this one in ignoreErrors?

File: src/View/Form/EntityContext.php [hunk 1]
```diff
@@ -534,10 +547,10 @@ protected function _getValidator(array $parts): Validator
     /**
      * Get the table instance from a property path
      *
-     * @param array|string|\Cake\Datasource\EntityInterface $parts Each one of the parts in a path for a field name
+     * @param string[]|string|\Cake\Datasource\EntityInterface $parts Each one of the parts in a path for a field name
      * @param bool $fallback Whether or not to fallback to the last found table
      *  when a non-existent field/property is being encountered.
-     * @return \Cake\ORM\Table|bool Table instance or false
+     * @return \Cake\ORM\Table|false Table instance or false
```
Comment (ADmad) [inline src/View/Form/EntityContext.php]: How about making it return `null` instead of `false` if no matching table is found?

Comment (dereuromark) [reply src/View/Form/EntityContext.php]: I agree with you. But can we still do that here in 4.x?

Comment (ADmad) [reply src/View/Form/EntityContext.php]: Sure can.

**Review by ADmad — APPROVED** [review]

**Pull request merged**